PhotoLocator/AutoTagViewModel.cs

```diff
@@ -139,15 +139,29 @@ protected bool SetProperty<T>(ref T field, T newValue, [CallerMemberName] string
         
         private IEnumerable<GpsTrace> LoadAdditionalGpsTraces()
         {
-            if (string.IsNullOrEmpty(TraceFilePath))
+            if (string.IsNullOrWhiteSpace(TraceFilePath))
                 return GpsTraces;
+
+            // Support Windows Copy Full Path string to be used by the user, Example: "C:\temp\file.jpeg"
+            if (TraceFilePath.Contains("\"")) TraceFilePath = TraceFilePath.Replace("\"", "");
+
+            if (!Path.Exists(TraceFilePath))
+                throw new DirectoryNotFoundException("Provided gps trace file or folder not found");
+
             var minDistance = TimeSpan.FromMinutes(MaxTimestampDifference);
             Directory.SetCurrentDirectory(Path.GetDirectoryName(_selectedItems.First().FullPath)!);
-            if (File.Exists(TraceFilePath))
-                return GpsTraces.Concat(GpsTrace.DecodeGpsTraceFile(TraceFilePath, minDistance));
-            return GpsTraces.
-                Concat(Directory.EnumerateFiles(TraceFilePath, "*.gpx").SelectMany(fileName => GpsTrace.DecodeGpsTraceFile(fileName, minDistance))).
-                Concat(Directory.EnumerateFiles(TraceFilePath, "*.kml").SelectMany(fileName => GpsTrace.DecodeGpsTraceFile(fileName, minDistance)));
+
```

Comment:
What is the difference between this code and the original version?

Reply:
You are right. At the moment this is _functionally_ identical. However the changed code is more explicit about what is happening. Which in my account makes it easier to support in the long run.